gtn/functions/shortest.cpp

```diff
@@ -192,7 +192,7 @@ Graph shortestPath(const Graph& g) {
   std::queue<int> computed;
   // List of in degrees for each node
   std::vector<int> degrees;
-  degrees.reserve(g.numNodes());
+  degrees.resize(g.numNodes());
```

Comment:
I think we can combine the two statement here to one
```
std::vector<int> degrees(g.numNodes());
```

Reply:
There are two different `degrees` vectors, but yeah I'll switch to the ctor